pkg/kubelet/dockershim/network/cni/cni_others.go

```diff
@@ -61,20 +64,52 @@ func (plugin *cniNetworkPlugin) platformInit() error {
 // TODO: Use the addToNetwork function to obtain the IP of the Pod. That will assume idempotent ADD call to the plugin.
 // Also fix the runtime's call to Status function to be done only in the case that the IP is lost, no need to do periodic calls
 func (plugin *cniNetworkPlugin) GetPodNetworkStatus(namespace string, name string, id kubecontainer.ContainerID) (*network.PodNetworkStatus, error) {
+
+	podIP := plugin.GetPodIP(id)
+	if podIP != "" {
+		klog.V(3).Infof("get pod ip %s from plugin", podIP)
+		return &network.PodNetworkStatus{IP: net.ParseIP(podIP)}, nil
+	}
+	cninetwork := plugin.getDefaultNetwork()
+	cniNet := cninetwork.CNIConfig
+	// TODO chaining cni?
+	netConf := cninetwork.NetworkConfig.Plugins[0]
 	netnsPath, err := plugin.host.GetNetNS(id.ID)
 	if err != nil {
 		return nil, fmt.Errorf("CNI failed to retrieve network namespace path: %v", err)
 	}
 	if netnsPath == "" {
 		return nil, fmt.Errorf("Cannot find the network namespace, skipping pod network status for container %q", id)
 	}
+	rt, err := plugin.buildCNIRuntimeConf(name, namespace, id, netnsPath, nil, nil)
+	if err != nil {
+		klog.Errorf("Error get pod network status when building cni runtime conf: %v", err)
+		return nil, err
+	}
+	res, err := cniNet.GetNetworkCachedResult(netConf, rt)
```

Comment:
Don't actually need to to through all the buildCNIRuntimeConf() and getting the NS and all that. It's enough to set the rt.ContainerID and rt.IfName (always "eth0" for kubelet) correctly.

And sorry about the "chaining cni?" comment stuff. You don't actually need to do any of this, just call GetNetworkListCachedResult() instead; it was an oversight that wasn't part of the interface but that has been fixed in https://github.com/containernetworking/cni/pull/669 so if you revendor you can grab simplify this code a lot.

Reply:
thanks, can I rely on GetNetworkListCachedResult will be in 0.7.0 libcni?

Reply:
0.7.0 was released before that change; 0.7.1 or later will have it.